### PR TITLE
adding server-ssh blueprint test

### PR DESCRIPTION
**Pull request by EarthmanT**

### CODE REVIEW

File: openstacksdk_plugin/utils.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+def set_ctx(_ctx):
+    if _ctx.type == RELATIONSHIP_INSTANCE:
+        if _ctx._context.get('is_target', True):
+            return _ctx.target
+        return _ctx.source
+    if _ctx.type != NODE_INSTANCE:
+        _ctx.logger.warn(
+            'CloudifyContext is neither {0} nor {1} type. '
+            'Falling back to {0}. However, this indicates a problem.')
```
Comment (mabuaisha) [inline openstacksdk_plugin/utils.py]: @EarthmanT  I think you are missing values for `{0}, {1}` what are the values that should be populated for those two ?

File: plugin.yaml [hunk 2]
```diff
@@ -796,8 +796,53 @@ node_types:
...
-  cloudify.openstack.server_connected_to_server_group:
+  cloudify.relationships.openstack.server_connected_to_server_group:
     derived_from: cloudify.relationships.connected_to
+
+  cloudify.relationships.openstack.server_connected_to_port:
+    derived_from: cloudify.relationships.connected_to
+    target_interfaces:
+      cloudify.interfaces.relationship_lifecycle:
+        postconfigure:
+          implementation: openstacksdk.openstacksdk_plugin.resources.network.port.update
```
Comment (mabuaisha) [inline plugin.yaml:814]: @EarthmanT We are sure at this moment that the server instance is created right ? I mean the device_id should be ready in order to passed it to the postconfigure action ?

File: plugin.yaml [hunk 3]
```diff
@@ -796,8 +796,53 @@ node_types:
...
+          inputs:
+            device_id:
+              default: ''
+
+  cloudify.relationships.openstack.port_connected_to_router:
+    derived_from: cloudify.relationships.connected_to
+    target_interfaces:
+      cloudify.interfaces.relationship_lifecycle:
+        postconfigure:
```
Comment (mabuaisha) [inline plugin.yaml:828]: The same here ?

File: test-requirements.txt [hunk 4]
```diff
@@ -5,4 +5,4 @@ mock>=1.0
 flake8
 cloudify>=4.0
 tox
-
+fabric
```
Comment (mabuaisha) [inline test-requirements.txt]: Are we going to use fabric 2 ?

**Review by mabuaisha — COMMENTED** [review]
Left some comments

File: openstacksdk_plugin/utils.py (hunk 1 above)
Comment (EarthmanT) [reply openstacksdk_plugin/utils.py]: woops

File: test-requirements.txt (hunk 4 above)
Comment (EarthmanT) [reply test-requirements.txt]: Good point. We actually do not need this.

**Pull request merged**